### PR TITLE
feat(core): journal sidebar conflict block

### PR DESCRIPTION
**Pull request by CatsJuice**

### CODE REVIEW

**Comment by CatsJuice** [comment]
Current dependencies on/for this PR:
* `canary`
  * **PR #5468** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5468?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #5477** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5477?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #5517** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5517?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
        * **PR #5549** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5549?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
          * **PR #5557** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5557?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
            * **PR #5558** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5558?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
              * **PR #5559** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5559?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                * **PR #5574** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5574?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
                  * **PR #5578** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5578?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
                    * **PR #5579** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/5579?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/toeverything/AFFiNE/5574?utm_source=stack-comment).

File: packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx [hunk 1]
```diff
@@ -223,6 +242,96 @@ const JournalDailyCountBlock = ({ workspace, page }: EditorExtensionProps) => {
...
+
+  if (pages.length <= 1) return null;
+
+  return (
+    <ConflictList
+      className={styles.journalConflictBlock}
+      pages={pages.slice(0, MAX_CONFLICT_COUNT)}
+      {...props}
+    >
+      {pages.length > MAX_CONFLICT_COUNT && (
```
Comment (pengx17) [inline packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.tsx]: don't use `&&` in render function. use ` ? xxx : null` instead

**Review by pengx17 — APPROVED** [review]

File: packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.css.ts [hunk 2]
```diff
@@ -151,4 +158,44 @@ export const pageItemLabel = style({
...
+    borderRadius: 4,
+    padding: '0px 8px',
+    fontSize: 'var(--affine-font-size-sm)',
+    display: 'flex',
+    alignItems: 'center',
+  },
+]);
+
+// override date-picker's active day when is not journal
+globalStyle(
```
Comment (joooye34) [inline packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.css.ts:194]: Use selectors? I think this kind of globalStyle usage is mislead.

Comment (CatsJuice) [reply packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.css.ts:194]: The date cell in global `AFFiNEDatePicker` is not customizable, but we need to override style only for journal. I'm not sure what u mean 'use selectors'?

Comment (joooye34) [reply packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.css.ts:194]:
```
export const journalPanel = style({
  'selectors': {
     '&[data-is-journal="false"] .react-datepicker__day[aria-selected="true"]': {
       backgroundColor: 'transparent',
       color: 'var(--affine-text-primary-color)',
       fontWeight: 500,
       border: '1px solid var(--affine-primary-color)',
     }
  }
})
```

Comment (joooye34) [reply packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.css.ts:194]: like this

Comment (joooye34) [reply packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.css.ts:194]: is it get invalid selectors error ？

**Review by joooye34 — APPROVED** [review]

File: packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.css.ts (hunk 2 above)
Comment (CatsJuice) [reply packages/frontend/core/src/pages/workspace/detail-page/editor-sidebar/extensions/journal.css.ts:194]: This syntax is not allowed by `@vanilla-extract/css`, and will cause error: `Style selectors must target the '&' character`

> https://vanilla-extract.style/documentation/styling/#complex-selectors
> To improve maintainability, each style block can only target a single element. To enforce this, all selectors must target the `&` character which is a reference to the current element.

**Pull request merged**